lib/services/services_linux.c

```diff
@@ -911,6 +911,28 @@ resources_os_list_ocf_agents(const char *provider)
     return result;
 }
 
+gboolean
+resources_os_ocf_agent_exists(const char *provider, const char *agent)
```

Comment:
Let's call this one services__ocf_agent_exists(). We have lots of legacy functions from before we tried to establish a consistent naming policy, so choosing examples is a hazardous process :) but that fits the current model.

Also, OCF agents are kept in a predictable file path, so I think it would be more efficient to implement this as a stat() call rather than list the whole directory. The format is OCF_ROOT_DIR "/resource.d/%s/%s", provider, agent.